### PR TITLE
✨ Added Hiven Presence

### PR DESCRIPTION
**Pull request by OfficialCRUGG**
This PR adds a Presence for Chat Platform Hiven.

Screenshots:
image
image
image

### CODE REVIEW

File: websites/H/Hiven/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "service": "Hiven",
+    "description": {
+        "en": "Hiven connects you to friends, fans and new exciting communties. From private, to public, to monetized groups - Hiven's where it happens."
+    },
+    "url": [
+        "hiven.io",
+        "app.hiven.io"
+    ],
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/uwQcgk5.jpg",
```
Comment (Bas950) [inline websites/H/Hiven/dist/metadata.json:15]: logo must be at least 512x512

**Review by Bas950 — CHANGES_REQUESTED** [review]
kinda low quality presence, can you add more to it?

**Comment by OfficialCRUGG** [comment]
Will update the logo for this one in a bit too.
As for content: I don't know what else there'd be to add. Except maybe show whether the user is in a voice call or not, but I currently have nobody to test this with. But once I'd have someone to test this with, I'd have added it anyways.

**Comment by OfficialCRUGG** [comment]
Just noticed there might be some more things I could add, though. Will do another commit shortly.

File: websites/H/Hiven/presence.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          }
+        ];
+      }
+    }
+
+    // Check if user is in settings
+    if (document.querySelector("div.sc-pjIPr.fZnZKj") && document.querySelector("div.sc-pjIPr.fZnZKj").textContent === "User Settings") {
+      presenceData.details = "Viewing their settings";
+      if (privacy) 
+        presenceData.state = undefined;
```
Comment (Slowlife01) [inline websites/H/Hiven/presence.ts:70]:
```suggestion
        delete presenceData.state;
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
format it a bit too many empty lines...

File: websites/H/Hiven/presence.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    // Check if user is in settings
+    if (document.querySelector("div.sc-pjIPr.fZnZKj") && document.querySelector("div.sc-pjIPr.fZnZKj").textContent === "User Settings") {
+      presenceData.details = "Viewing their settings";
+      if (privacy) 
+        delete presenceData.state;
+      else 
+        presenceData.state = document.querySelector("div.sc-pbWVv.hTvDIL").textContent;
+
```
Comment (Slowlife01) [inline websites/H/Hiven/presence.ts:71]:
```suggestion
```

File: websites/H/Hiven/presence.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        presenceData.state = document.querySelector("div.sc-pbWVv.hTvDIL").textContent;
+      
+    }
+
+    // Check if user is viewing Settings of a House
+    if (document.querySelector("div.sc-psfJB.efwVkl")) {
+      presenceData.details = "Viewing a House's settings";
+      if (!privacy) 
+        presenceData.state = document.querySelector("div.sc-psfJB.efwVkl").children[2].textContent;
+
```
Comment (Slowlife01) [inline websites/H/Hiven/presence.ts:79]:
```suggestion
```

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**